src/movies.js

```diff
@@ -1,25 +1,90 @@
+
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors(moviesArray) {}
+  
+function getAllDirectors(moviesArray){
+    return  moviesArray.map(function(element){
+       return element.director
+    }) 
+}
+getAllDirectors(movies)
+
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies(moviesArray) {}
+function howManyMovies(moviesArray) { 
+    return moviesArray.filter(function(element){
+       return  element.director==='Steven Spielberg' && element.genre.includes('Drama')===true
+    }).length
+   
+ } 
+(howManyMovies(movies))
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage(moviesArray) {}
+function scoresAverage(moviesArray) {
+    if(moviesArray.length===0){
+        return 0
+    }
+   
+    const scoresAv=(moviesArray.reduce(function(sum, currentValue){
+        return sum+currentValue.score
```

Comment:
Here we need to check the type of score, sometimes they dont pass it so wee need to make sure its a number

```suggestion
       if(typeof currentValue.score==="number"){
return currentValue.score +sum
    }
      return sum
```